File: ServerApp/Karuna/Karuna/src/main/java/com/karuna/services/ReceiverService.java
package com.karuna.services;

import java.util.List;

import com.karuna.dto.LoginDto;
import com.karuna.dto.LogoutDto;
import com.karuna.dto.RegisterReceiverDto;
import com.karuna.dto.RequestDto;
import com.karuna.entity.Donor;
import com.karuna.entity.Item;
import com.karuna.entity.Receiver;
import com.karuna.entity.Request;

public interface ReceiverService {
	
    Receiver registerReceiver(RegisterReceiverDto newReceiver);
	
	Receiver login(LoginDto loginDto);
	
	Item receive(Item item);
	
	String logout(LogoutDto logoutDto);
	
	String deleteAccount(Long receiverId);
	
	Receiver updateReceiver(RegisterReceiverDto updatedReceiver);

	List<Item> viewItems();
	
	List<Item> viewHistory(Receiver receiver);
	
	List<Donor> viewDonors();
	
	Request createRequest(RequestDto request);

	
	
	
	

}

File: ClientApp/karuna/src/services/ReceiverService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080';

class ReceiverService {
  registerReceiver(receiver) {
    return axios.post(`${API_URL}/receiver/register`, receiver);
  }

  viewItems() {
    return axios.get(`${API_URL}/receiver/items`);
  }

  deleteAccount(receiverId) {
    return axios.delete(`${API_URL}/receiver/${receiverId}`);
  }

  updateReceiver(updatedReceiver) {
    return axios.put(`${API_URL}/receiver/profile`, updatedReceiver);
  }

  login(dto) {
    return axios.post(`${API_URL}/receiver/login`, dto);
  }

  createRequest(reqDto) {
    return axios.post(`${API_URL}/receiver/request`, reqDto);
  }

  viewDonors() {
    return axios.get(`${API_URL}/receiver/donors`);
  }

  viewHistory(receiverId) {
    return axios.get(`${API_URL}/receiver/history/${receiverId}`);
  }

  receive(itemId) {
    return axios.put(`${API_URL}/receiver/${itemId}`);
  }

  logout(receiverId) {
    return axios.get(`${API_URL}/receiver/logout/${receiverId}`);
  }
}

export default new ReceiverService();
